### PR TITLE
Switch to pnpm, upgrade ember-cli-babel to 8.2

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Of note, I disabled the try scenarios for 
- release
- beta
- canary

I believe they are failing due to recent changes in ember-source, but in any case, they are very likely unrelated to the following plan:

Planned PRs:
- pnpm
- add release-plan
- release things on main, if any (there is at least a patch to do)
- extract test-app
- fix type-imports
- find old code and remove it
- native library conversion
  - resolves
    - https://github.com/emberjs/ember-test-helpers/issues/1448 (via not needing ember-cli-babel)  

This process will help determine what type of semver bump the next version will be

### CODE REVIEW

File: addon/addon-test-support/@ember/test-helpers/global.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ export default (() => {
     return self;
   } else if (typeof window !== 'undefined') {
     return window;
+    // Global doesn't exist, but older nodes might have it (relevant for fastboot)
```
Comment (NullVoxPopuli) [inline addon/addon-test-support/@ember/test-helpers/global.ts:8]: in a future PR, I may replace this whole function with `globalThis`

**Comment by NullVoxPopuli** [comment]
5.4 and 5.8 _should_ pass as soon as I adjust the ember-try config.

release, beta, and canary are expected to fail, as I think they actually have a bug around waiters

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -50,30 +72,32 @@ jobs:
           - ember-lts-4.12
           - ember-lts-5.4
           - ember-lts-5.8
-          - ember-release
-          - ember-beta
-          - ember-canary
+          # All of these are failing with the waiter tests (needs to be debugged separately)
```
Comment (kategengler) [inline .github/workflows/ci-build.yml:75]: Can you open an issue so this isn't forgotten?

**Review by kategengler — APPROVED** [review]

File: .github/workflows/ci-build.yml (hunk 2 above)
Comment (NullVoxPopuli) [reply .github/workflows/ci-build.yml:75]: there is this one: https://github.com/emberjs/ember.js/issues/20724

Comment (NullVoxPopuli) [reply .github/workflows/ci-build.yml:75]: https://github.com/emberjs/ember-test-helpers/issues/1465

**Pull request merged**